public/tier1/convar.h

```
@@ -464,251 +337,244 @@ friend class ConCommandHandle;
 
 	union
 	{
-		FnCommandCompletionCallback	m_fnCompletionCallback;
-		ICommandCompletionCallback* m_pCommandCompletionCallback;
+		FnCommandCompletionCallback	fnCompletionCallback;
+		ICommandCompletionCallback* pCommandCompletionCallback;
 	};
 
-	bool m_bHasCompletionCallback : 1;
-	bool m_bUsingCommandCompletionInterface : 1;
-	
-	ConCommandRefAbstract *m_pReference;
-};
+	bool has_complitioncallback;
+	bool is_interface;
 
+	ConVarHandle* refHandle;
+};
+static_assert(sizeof(ConCommandCreation_t) == 0x40, "ConCommandCreation_t is of the wrong size!");
 
-//-----------------------------------------------------------------------------
-// Purpose: A console variable
-//-----------------------------------------------------------------------------
-class ConVar
+// TO-DO: Remove this...
+class ConCommandBase
 {
-friend class CCvar;
-friend class ConVarRef;
-friend class SplitScreenConVarRef;
+public:
+	inline const char* GetName() { return "Please remove the ConCommandBase class..."; }
+}; // For metamod compatibility only!!!!

```

Wouldn't it be better to just remove it completely instead of faking it? As that'll let the compiler to yell on missing stuff instead of silently fail?

---

I don't want to break metamod compilation with this PR. But if @psychonic is fine with it, then lets do it

---

Go for it. MM:S needs to follow what we make of the SDK, not the other way around.